app/decorators/prosecution_case_decorator.rb

```diff
@@ -1,18 +1,16 @@
 # frozen_string_literal: true
 
 class ProsecutionCaseDecorator < BaseDecorator
+  attr_accessor :column, :direction
+
   def hearings
     @hearings ||= decorate_all(object.hearings)
   end
 
-  def hearings_by_datetime
-    hearings.sort_by { |h| h.hearing_days.map(&:to_datetime) }
-  end
-
-  def hearings_with_day_by_datetime
+  def sorted_hearings_with_day
     Enumerator.new do |enum|
-      hearings_by_datetime.each do |hearing|
-        hearing.hearing_days.map(&:to_datetime).sort.each do |day|
+      sorter.sorted_hearings.each do |hearing|
+        sorter.sorted_hearing(hearing).each do |day|
```

Comment:
although more verbose `sorted_hearing_days` is more declarative

Reply:
👍 I have amended sorted_hearing to sorted_hearing_days